yarn-project/txe/src/oracle/txe_oracle.ts

```diff
@@ -132,7 +132,7 @@ export class TXE implements TypedOracle {
   ) {
     this.noteCache = new ExecutionNoteCache(this.getTxRequestHash());
     this.contractDataOracle = new ContractDataOracle(txeDatabase);
-    this.contractAddress = AztecAddress.random();
+    this.contractAddress = AztecAddress.ZERO;
```

Comment:
Would this mean that by default in a test a user would get zero contract address?

If yes then that seems like a problem as that is quite a different thing from random address and could cause confusion by devs.

Why not just use a private constructor and a static createInstance function pattern here?

Reply:
TBH, I thought this would actually be better. Zero is in fact an invalid address, but that means you never called `env.impersonate(contract_address)` in your test, which would mean you're performing a tx from limbo. We should even force failure in this case, since that would never happen in real life (you either call from an account contract, or from another contract)

Reply:
It's not fine indeed 😅 . Adding an init method!

Zero is an unfortunate number. There's a test that relies on not initializing a contract address at all (which I guess was serialized as a 0)